### ISSUE TITLE
Hostname, SPI pins, and other improvments

### ISSUE THREAD

**Issue by chrishoage** [wuspy/portal_calendar#8]
I hacked on your project this weekend. 

I made the following improvements: 

1. Allow a hostname (I wanted something more recognizable in my network, I have a ton of ESP devices :sweat_smile:) 
2. Allow specifying SPI pins (when using HSPI on my ESP32S2 none of the pins are defined - and need to be manually defined)
3. Toggle between chamber and weather display using the GPIO 0 button (e.g. boot button) 

https://user-images.githubusercontent.com/1056886/220210141-7406e1ac-9b14-4275-9178-c0dba52aa30b.mp4

See the changes here in a comparison:

https://github.com/wuspy/portal_calendar/compare/master...chrishoage:portal_calendar:hoage/improvements

If you would like any of these to be upstream let me know! I'd be happy to cherry pick the commits to branches and open PRs. 

Thank you for your wonderful project :slightly_smiling_face: (sorry for opening an issue - github discussions don't seem to be enabled!)

**Comment by wuspy** [comment]
I left a couple comments on the commits. Configurable hostname and SPI pins sounds great, if you wanna open a PR for those I'll merge them. I also think switching between icons & weather is a really good idea, but open a separate PR for that since it's a bigger change that I want to look at and test a bit more.

**Comment by chrishoage** [comment]
@wuspy Sounds good! I opened the SPI and Hostname PRs. I want to rework the toggle one. Right now I'm relying on EEPROM but I discovered `RTC_DATA_ATTR` which is what I originally wanted. 

An aside: Depending on the outcome of https://github.com/wuspy/portal_calendar/pull/11 it might be worth renaming `config.h` to `example_config.h` and adding `config.h` to `.gitignore.` (possibly also a `defaults.h` with some `ifndef` to set defaults for weather and other options) 

It's a dance to keep my local SPI / wifi / other config changes from getting commited - in particular when working on config.h itself.
